src/main/java/org/generationcp/middleware/service/api/dataset/ObservationUnitsSearchDTO.java

```diff
@@ -52,6 +52,7 @@ public class Filter {
 		private Set<Integer> filteredNdExperimentIds;
 		private Map<String, String> variableTypeMap;
 		private Integer variableId;
+		private Boolean variableHasValue;
```

Comment:
@mderamos what exactly means variableHasValue? has value? has cValue_id? hasPhenotype?

Reply:
Hi @clarysabel , 
**what exactly means variableHasValue? has value? has cValue_id? hasPhenotype?**
it's either has value or has draftValue depending on the draft mode. It's used for the prompt message showing how many observations will be affected with the delete variable values batch action. Only those with values(or draft values depending on the draft mode) should be filtered. It's also used for the actual deletion method so that the query will only return the necessary observations. 

**what happens if draftMode is TRUE and variableHasValue is TRUE?**
if the draftmode is true, it already filters the observations with non-null draft value so it will just ignore the _variableHasValue _ filter.

**It may be possible to set this value in the REST service and not in the frontend**
Done addressing this.